libid/complex.cpp

```diff
@@ -0,0 +1,536 @@
+// SPDX-License-Identifier: GPL-3.0-only
+//
+// Complex.cpp: interface for the Complex class.
+//
+//////////////////////////////////////////////////////////////////////
+#include "Complex.h"
+
+//Complex::Complex(void)
+//    {
+//    }
+
+Complex::~Complex(void)
+    {
+    }
+
+Complex Complex::operator=(const double & value)	// Assignment to double Operator
+    {
+    x = value;
+    y = 0.0;
+    return *this;
+    }
+
+Complex Complex::operator=(const Complex & Cmplx1)	// Assignment Operator
+    {
+    x = Cmplx1.x;
+    y = Cmplx1.y;
+    return *this;
+    }
+
+// new stuff
+Complex Complex::operator+=(const Complex & Cmplx1)
```

Comment:
Returns `Complex &`, not `Complex`.

Same for `operator-=`, `operator*=`,  and similar operators.

See https://en.cppreference.com/w/cpp/language/operators for details.